### ISSUE TITLE
API breaks when moving to ORJSONResponse due to the use of integers as response keys.

### ISSUE THREAD

**Issue by hellocoldworld** [tiangolo/fastapi#2934]
### First check

* 

- [ DONE] I added a very descriptive title to this issue.

* [ DONE] I used the GitHub search to find a similar issue and didn't find it.
* [ DONE] I searched the FastAPI documentation, with the integrated search.
* [ DONE] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [ DONE] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ DONE] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ DONE] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ DONE] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ YES] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.
 
There is an issue when moving from default JSON responses to ORJSONResponse. When returning a Dict that uses integers as keys, default json responses will cast them to string, in order to produce valid json output. ORJSON will refuse to serialize this response.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

This works:

```Python
from fastapi import FastAPI
from fastapi.responses import ORJSONResponse

app = FastAPI()

@app.get("/")
async def dict():
    return {1:2}

if __name__ == "__main__":
    import uvicorn
    uvicorn.run("service:app", port=8001, workers=1)
```

But this fails:

```Python
from fastapi import FastAPI
from fastapi.responses import ORJSONResponse

app = FastAPI(default_response_class=ORJSONResponse,)

@app.get("/")
async def dict():
    return {1:2}

if __name__ == "__main__":
    import uvicorn
    uvicorn.run("service:app", port=8001, workers=1)
```

You can test both behaviours using swagger UI.

Also, I don't know if there's something we can do here, since it seems to be the way ORJSON was defined, but I think we can at least add a comment about it in the docs.

### Environment

* OS: [e.g. Linux / Windows / macOS]: Tested on Ubuntu MATE 18.04, MacOS, and alpine linux
* FastAPI Version [e.g. 0.3.0]: 0.63.0 in all cases
* Python version: 3.7, 3.8 and 3.9

**Comment by valentin994** [comment]
As keys have to be strings I don't really see a reason to put that in the docs. If you change the response to {"1": 2} it's gonna work like expected. That's how JSON is defined in general. But it is true that it will work the other way around as you specified. So maybe make it not work either way when the key is not a string or cast it to str when using orj. Interesting find though :D

**Comment by juntatalor** [comment]
orjson by default does not support integer keys, but it is possible to control this behavior with specail flags (https://github.com/ijl/orjson#opt_non_str_keys)

You could subclassJSONResponse like this:
```
import orjson

class ORJSONResponseCustom(JSONResponse):

    option = orjson.OPT_NON_STR_KEYS

    def render(self, content: Any) -> bytes:
        return orjson.dumps(content, option=self.option)
```

**Comment by hellocoldworld** [comment]
Yes I understand both of your comments.I was surprised by this behaviour, so my question was if this was worthy of a mention on the docs, (or even unify the behaviour in fastapi).

**Comment by ycd** [comment]
A lot of discussions happened on this topic, see:

#1137 
#1165
#1846 (A PR to add this feature)
#2584 (Duplicate of #1846 but has the same motivation)

**Comment by raphaelauv** [comment]
@ycd is right , could you please close the issue , thank

**Comment by tiangolo** [comment]
Thanks for the interest @hellocoldworld and thanks a lot for the help @ycd and @raphaelauv! :bow: 

I added a section to the docs explaining how to create a custom response: https://fastapi.tiangolo.com/advanced/custom-response/#custom-response-class
